gcloud-java-compute/src/main/java/com/google/gcloud/compute/Region.java

```diff
@@ -0,0 +1,319 @@
+          return Region.fromPb(pb);
+        }
+      };
+  static final Function<Region, com.google.api.services.compute.model.Region> TO_PB_FUNCTION =
+      new Function<Region, com.google.api.services.compute.model.Region>() {
+        @Override
+        public com.google.api.services.compute.model.Region apply(Region region) {
+          return region.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = -3578710133393645135L;
+
+  /**
+   * Status of the region.
+   */
+  public enum Status {
+    UP,
+    DOWN
+  }
+
+  /**
+   * A quota assigned to this region.
+   */
+  public static class Quota implements Serializable {
+
+    static final Function<com.google.api.services.compute.model.Quota, Quota> FROM_PB_FUNCTION =
+        new Function<com.google.api.services.compute.model.Quota, Quota>() {
+          @Override
+          public Quota apply(com.google.api.services.compute.model.Quota pb) {
+            return Quota.fromPb(pb);
+          }
+        };
+    static final Function<Quota, com.google.api.services.compute.model.Quota> TO_PB_FUNCTION =
+        new Function<Quota, com.google.api.services.compute.model.Quota>() {
+          @Override
+          public com.google.api.services.compute.model.Quota apply(Quota quota) {
+            return quota.toPb();
+          }
+        };
+    private static final long serialVersionUID = -4357118665133226338L;
+
+    private final String metric;
+    private final double limit;
+    private final double usage;
+
+    /**
+     * Returns a region quota object.
+     */
+    Quota(String metric, double limit, double usage) {
+      this.metric = metric;
+      this.limit = limit;
+      this.usage = usage;
+    }
+
+    public String metric() {
+      return metric;
+    }
+
+    public double limit() {
+      return limit;
+    }
+
+    public double usage() {
+      return usage;
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("metric", metric)
+          .add("limit", limit)
+          .add("usage", usage)
+          .toString();
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(metric, limit, usage);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (!(obj instanceof Quota)) {
+        return false;
+      }
+      Quota other = (Quota) obj;
+      return Objects.equals(metric, other.metric)
+          && Objects.equals(limit, other.limit)
+          && Objects.equals(usage, other.usage);
+    }
+
+    com.google.api.services.compute.model.Quota toPb() {
+      return new com.google.api.services.compute.model.Quota()
+          .setMetric(metric)
+          .setLimit(limit)
+          .setUsage(usage);
+    }
+
+    static Quota fromPb(com.google.api.services.compute.model.Quota quotaPb) {
+      return new Quota(quotaPb.getMetric(), quotaPb.getLimit(), quotaPb.getUsage());
+    }
+  }
+
+  private final RegionId regionId;
```

Comment:
Nit: here and elsewhere, can we move these variables to the beginning of the class?